src/main.rs

```diff
@@ -181,16 +182,17 @@ fn fetch_project(matches: &ArgMatches, config: &Config) -> Result<String, String
 fn task_create(matches: &ArgMatches) -> Result<String, String> {
     let config = fetch_config(matches)?;
     let content = fetch_string(matches, "content", "Content")?;
+    let priority = fetch_string(matches, "priority", "Priority")?;
```

Comment:
Actually, can we make this a select? As there are only 4 valid options.

Reply:
I don't really sure that I have understood you correctly.
I made some changes, can you tell is it what you had in mind?

Basically if flag wasn't provided - there is prompt for priority now when you use `task create`
I will clean code in next commit (maybe encapsulate some logic in Priority struct, create separate file for it etc)

Reply:
Yeah, absolutely. The modification I am hoping for is to use [Config::select_input)[https://github.com/alanvardy/tod/blob/5b2fd5b060510d6a1786d33495a7beb035569052/src/config.rs#L246] instead of `Config::get_input` so that the user can select the priority (but can still just type the number too)

Reply:
As I understand I have basically the same thing right now

```rust
#[cfg(not(tarpaulin_include))]
fn task_create(matches: &ArgMatches) -> Result<String, String> {
    use inquire::Select;
    use items::Priority;

    let config = fetch_config(matches)?;
    let content = fetch_string(matches, "content", "Content")?;
    let priority = match Priority::get_from_matches(&matches) {
        Some(value) => value,
        None => {
            let options = vec![
                Priority::None,
                Priority::Low,
                Priority::Medium,
                Priority::High,
            ];
            Select::new(
                "Choose a priority that should be assigned to task:",
                options,
            )
            .prompt()
            .map_err(|e| e.to_string())
            .expect("Failed to create option list of priorities")
        }
    };

    let project = fetch_project(matches, &config)?;

    projects::add_item_to_project(&config, content, &project, priority)
}

```

you want me to reuse Config::select_input in this situation to avoid dublication? But I can't give it a vec of Priority, only a vec of strings.
I can make it generic so it can take a type that implements Display, like this

```rust
pub fn select_input<T: Display>(desc: &str, options: Vec<T>) -> Result<T, String> {
    Select::new(desc, options)
        .prompt()
        .map_err(|e| e.to_string())
}

```

but the part with test data must go or it will not compile (at least I don't know how to make it compile with it)
```rust
    if cfg!(test) {
        return Ok(String::from("Africa/Asmera"));
    }
```

Maybe I missing something =)

Reply:
I like your implementation with the generic type that implements display. What if we got the select to return the first item of the Vector when `config!(test)` That way the function would always return the correct type.

Reply:
Hi,
I have commited your suggestion. But now there are problems with config_tests tests because of the fact that the method returns different timezone (first from the list).
I fix one test and another becomes broken =) They are hard to follow. There is need for some juggling with data.
 
 I would be grateful if you can help me to resolve them =)

As I understand `if cfg!(test)` there for "mocking the user input"?
It looks somewhat messy and hard to follow in general, wonder if there is a better or simpler way to do it.

Reply:
I pulled in your branch and pushed it up to this repo on this PR here: https://github.com/alanvardy/tod/pull/351

I agree that there's lots of improvement needed around the testing, I'm kinda figuring it out as I go and don't have a solid grasp on what standard practice is in the Rust community :+1: 

That's right, I'm just wanting to mock user input, because without it the test would just fail on CI

Closing this PR here so we can continue with the new one.